### PR TITLE
app-dicts/dictd-*: EAPI bumps, new eclass

### PR DESCRIPTION
**Pull request by hanetzer**
As most of these ebuilds are essentially identical, I combined their common elements
into `dictd.eclass`. All ebuilds build tested; A few were renamed (mostly `YYYY.MM.DD`
`${PV}` to `YYYYMMDD`) in order to simplify their internal logic, so I had to use repoman's
--straight-to-stable flag.

### CODE REVIEW

**Comment by hanetzer** [comment]
Don't merge; I have some work I'm doing on this to streamline it more. Also should send
the eclass changes to the ml.

**Comment by hanetzer** [comment]
hmm... gentoo/gentoo-gitmig-20150809-draft@8c10135d6ba708afdf1cd267b845ae9007dbd9c5 seems to be what introduced the usage of
`$(get_libdir)` but that honestly does not look like it should be done, considering app-text/dictd's
dictd.confd file hardcodes `/usr/lib/dict`...

**Comment by mgorny** [comment]
What's the status here? I see a lot of conflicts.

**Comment by hanetzer** [comment]
@mgorny was your explicit EAPI=0 commit that did it, lol. Missed that.

**Comment by flappyports** [comment]
@mgorny, you OK with this eclass?

File: eclass/dict.eclass [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+RDEPEND="app-text/dictd"
+
+S="${WORKDIR}"
+
+# @FUNCTION: dict_src_install
+# @DESCRIPTION:
+# The dict src_install function, which is exported
+dict_src_install() {
+	insinto /usr/share/dict
+	for dict in "${DICTS[@]}"; do
```
Comment (mgorny) [inline eclass/dict.eclass]: You can avoid the loop and multiple `doins` calls via `doins ${DICTS[@]/%/.dict.dz} ...` (and the same for `.index`). `%` is bash's equivalent of regexp end-of-string `$`.

Comment (hanetzer) [reply eclass/dict.eclass]: Oh that's a cool trick indeed. Thanks for that.

**Comment by flappyports** [comment]
@hanetzer Have you sent this to -dev ML for review?

**Comment by hanetzer** [comment]
@flappyports there was one iteration of it, kinda poorly done, where me and
@ulm had a bit of back and forth on it. I can resend it and get the discussion
going again.

**Comment by mgorny** [comment]
Please do. Run it through -dev until you get reasonable ack (or no new comments for a few days), then get back to us.

File: eclass/dict.eclass [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+# @ECLASS-VARIABLE: DICTS
+# @DESCRIPTION:
+# Array of dictionary file names (foo.dict.dz and foo.index) to be installed
+# during dict_src_install
+
+if [[ -z ${_DICT_ECLASS} ]]; then
+_DICT_ECLASS=1
+
+case ${EAPI:-0} in
+	6) ;;
```
Comment (SoapGentoo) [inline eclass/dict.eclass]: mayeb also enable EAPI 7

Comment (hanetzer) [reply eclass/dict.eclass]: Ah yeah. I was working on this before it was acceptable upstream. Thanks for that.

**Pull request closed without merging**